packages/reactive-element/src/reactive-element.ts

```diff
@@ -737,6 +763,31 @@ export abstract class ReactiveElement
     return this.elementProperties.get(name) ?? defaultPropertyDeclaration;
   }
 
+  // This is a finalization step, but it needs to be separate from `finalize`
```

Comment:
I don't see why this must be separate from `finalize` and doing so adds complexity. 

1. doesn't need to be separate: although `finalize` can be called more than once, it immediately returns if it's already done its work. 

2. complexity: calling finalize also ensures that superclasses have also been finalized, and separating this metadata from that means that a superclass may not have metadata. I'm not sure if that matters or not.

Reply:
Talked offline: this needs to happen later than finalize(), which can be called early in addInitializer() and createProperty(). finalize() should probably be called prepare() in those cases.